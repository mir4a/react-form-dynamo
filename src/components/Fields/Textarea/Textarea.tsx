import React, { memo, useState, useCallback, ChangeEvent } from 'react';
import Label from '../Label';
import classNames from 'classnames';
import './Textarea.css';

interface Props {
  intitialValue?: string;
  label: string;
  name?: string;
  onChange?: Function;
  className?: string;
}

export default memo(function TextareaField(props: Props) {
  const { intitialValue, label, name, onChange, className } = props;
  const [value, setValue] = useState(intitialValue);
  const textareaClassNames = classNames('Textarea', className);
  const handleOnChange = useCallback(
    (event: ChangeEvent<HTMLTextAreaElement>) => {
      const value = event.target.value;
      setValue(value);
      onChange && onChange(value);
    },
    [setValue, onChange],
  );

  return (
    <>
      <Label label={label} forId={name} />
      <textarea
        className={textareaClassNames}
        id={name}
        name={name}
        onChange={handleOnChange}
        value={value}
      />
    </>
  );
});
