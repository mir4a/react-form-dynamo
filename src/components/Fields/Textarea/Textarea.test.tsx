import React from 'react';
import renderer from 'react-test-renderer';
import Textarea from './Textarea';

it('renders correctly', () => {
  const tree = renderer
    .create(<Textarea label="A" onChange={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
