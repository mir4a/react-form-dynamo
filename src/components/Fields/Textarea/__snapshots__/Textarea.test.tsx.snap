// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <label
    className="Label"
  >
    <span
      className="Label-text"
    >
      A
    </span>
  </label>,
  <textarea
    className="Textarea"
    onChange={[Function]}
  />,
]
`;
